tests/cpp/test_multidevice_pipeline.cpp

```diff
@@ -389,20 +391,14 @@ TEST_P(PipelineTestStagedReduction, StagedReduction) {
 
   FusionGuard fg(fusion.get());
   TensorView* tv0 = makeConcreteTensor(unsharded_input_sizes);
+  auto mesh = DeviceMesh::createForNumDevices(num_devices);
+  tv0->setDeviceMesh(mesh);
+  tv0->axis(0)->parallelize(ParallelType::DIDx);
```

Comment:
I'm not up to date with how tensors should get device parallelization. Could you tell me why the previous one fails and how this change fixes it?

Reply:
I pushed some new commits to make the fix obvious. PTAL